apps/dashboard/app/models/batch_connect/app.rb

```diff
@@ -238,6 +245,19 @@ def build_session_context
       BatchConnect::SessionContext.new(attributes, form_config.fetch(:cacheable, nil))
     end
 
+    def templates
+      @templates ||= begin
```

Comment:
I am wondering if this logic belongs here. It seems better suited to the controller.

Reply:
Moved the widgets rendering into `_form.html.erb` and this into the session contexts controller.